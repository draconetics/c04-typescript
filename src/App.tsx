import React, { Component, useEffect } from "react";
import {
  BrowserRouter,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import TodoComponent from "./components/TodoComponent";
import NoteListComponent from "./components/NoteListComponent/";
import MenuComponent from "./components/MenuComponent"
import LoginForm from "./pages/LoginForm"
import RegisterForm from "./pages/RegisterForm"

import Home from "./pages/Home"
import NotFoundPageComponent from "./pages/NotFoundPageComponent";

const  PrivateRoute: React.FC<{
  component: any;
  path: string;
  exact: boolean;
}> = ({path, exact, component,...rest}) => {
  console.log('private route');
  const data = {
    pathname: "/home",
    state: { 
      message: 'You should login or Register.',
      type:'alert-info'
    }
  }
  const condition = localStorage.getItem('token')!==null;
  return  condition ? (<Route {...rest} path={path} exact={exact} component={component} />) : 
    (<Redirect  to={data}  />);
};

const App: React.FC = () => {

  return (
    <React.Fragment>
      <BrowserRouter>
        <MenuComponent></MenuComponent>
        <Switch>
            <Route exact path={["/","/home"]} component={Home}></Route>
            <PrivateRoute path={"/notelist"} component={NoteListComponent} exact />
            <PrivateRoute path={"/todolist"} component={TodoComponent} exact />
            <Route path="/login" component={(localStorage.getItem('token')===null)?LoginForm:Home}></Route>
            <Route path="/register" component={(localStorage.getItem('token')===null)?RegisterForm:Home}></Route>
            <Route path='/404' component={NotFoundPageComponent} />
            <Redirect from='*' to='/404' />
        </Switch>
      </BrowserRouter>
      
    </React.Fragment>
  );
};
export default App;