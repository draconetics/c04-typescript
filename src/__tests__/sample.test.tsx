import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

/*
test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Add Todo/i);
  expect(linkElement).toBeInTheDocument();
});

*/
describe("#App component",()=>{
  it('show App component', (done) => {
    const component = shallow(<App />);  
    expect(component).toMatchSnapshot();
    done();
  });
})