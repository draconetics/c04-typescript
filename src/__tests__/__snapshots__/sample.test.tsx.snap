// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#App component show App component 1`] = `
<Fragment>
  <BrowserRouter>
    <MenuComponent />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path={
          Array [
            "/",
            "/home",
          ]
        }
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/notelist"
      />
      <Route
        component={[Function]}
        path="/todolist"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/register"
      />
      <Route
        component={[Function]}
        path="/404"
      />
      <Redirect
        from="*"
        to="/404"
      />
    </Switch>
  </BrowserRouter>
</Fragment>
`;
